pseudomonas_analysis/nbconverted/explore_expression_data.py

```diff
@@ -0,0 +1,634 @@
+    f"DE_stats_template_data_{project_id}_real.txt"
+)
+
+template_DE_stats = pd.read_csv(
+    template_DE_stats_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+selected = template_DE_stats[(template_DE_stats[pval_name]<0.01) & (abs(template_DE_stats[logFC_name])>1)]
+print(selected.shape)
+
+
+# In[6]:
+
+
+plot.make_volcano_plot_template(
+    template_DE_stats_filename,
+    project_id,
+    pval_name,
+    logFC_name
+)
+
+
+# In[7]:
+
+
+simulated_DE_stats_dir = os.path.join(local_dir, "DE_stats")
+
+plot.make_volcano_plot_simulated(
+    simulated_DE_stats_dir,
+    project_id,
+    pval_name,
+    logFC_name,
+    num_simulated,
+    5,
+    5,
+    20,
+    15
+)
+
+
+# ## Explore flat volcano plots
+# 
+# Why are some volcano plots flat?
+
+# In[8]:
+
+
+# Lets look at the expression data for the volcano plot with DEGs
+simulated_expression_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_11.txt"
+)
+
+simulated_expression = pd.read_csv(
+    simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+simulated_expression
+
+
+# In[9]:
+
+
+sns.distplot(simulated_expression.loc["GSM822708_wtLB_A.CEL"])
+sns.distplot(simulated_expression.loc["GSM822712_delta_cbrBLB_A.CEL"])
+
+
+# In[10]:
+
+
+# Heatmap visualization
+sns.heatmap(simulated_expression.iloc[:,10:20])
+
+
+# In[11]:
+
+
+# Lets look at the expression data for the flat volcano plot
+simulated_expression_filename = os.path.join(
+    local_dir,
+    "pseudo_experiment",
+    f"selected_simulated_data_{project_id}_5.txt"
+)
+
+flat_simulated_expression = pd.read_csv(
+    simulated_expression_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+flat_simulated_expression
+
+
+# In[12]:
+
+
+sns.distplot(flat_simulated_expression.loc["GSM822708_wtLB_A.CEL"])
+sns.distplot(flat_simulated_expression.loc["GSM822712_delta_cbrBLB_A.CEL"])
+
+
+# In[13]:
+
+
+# Heatmap visualization
+sns.heatmap(flat_simulated_expression.iloc[:,10:20])
+
+
+# In[14]:
+
+
+# Lets look at the DE stats associated with this flat volcano plot
+simulated_DE_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_simulated_data_{project_id}_5.txt"
+)
+
+flat_simulated_DE = pd.read_csv(
+    simulated_DE_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+flat_simulated_DE
+
+
+# In[15]:
+
+
+# Lets look at the DE stats associated with DEGs volcano plot
+simulated_DE_filename = os.path.join(
+    local_dir,
+    "DE_stats",
+    f"DE_stats_simulated_data_{project_id}_11.txt"
+)
+
+simulated_DE = pd.read_csv(
+    simulated_DE_filename, 
+    sep="\t", 
+    header=0, 
+    index_col=0
+)
+
+simulated_DE
+
+
+# Flat volcano plots are due to samples between the two groups are very similar, so the log2 fold change is very small 10^-1 for each gene and so the adjusted p-values are nearly the same for each gene. Another artifact of the VAE variance shrinkage.
```

Comment:
Everything above here was moved from `pseudomonas_analysis/viz_expression_data.ipynb`, right?

Reply:
Yes. Sorry for the confusion